src/storage/exec/RelNode.h

```diff
@@ -68,7 +68,7 @@ class RelNode {
 
   void addDependency(RelNode<T>* dep) {
     dependencies_.emplace_back(dep);
-    dep->hasDependents_ = true;
+    dep->isRoot_ = false;
```

Comment:
`hasDependents_` indicates that other nodes depend on this node.
This node must be executed first, and then the other can be executed.